api/src/npm/tarball.rs

```diff
@@ -106,155 +108,167 @@ pub async fn create_npm_tarball<'a>(
-              .iter()
-              .map(|d| match d.range() {
-                Some(range) => {
-                  format!("{}, at {}@{}", d, range.specifier, range.range.start)
-                }
-                None => format!("{}, at {}", d, d.specifier()),
-              })
-              .collect::<Vec<_>>()
-              .join(", ");
-            info!(
-              "Npm dts generation @{}/{}@{}: {}",
-              scope, package, version, message
-            );
+        if js.fast_check_module().is_some() {
+          let declaration_specifier = rewrite_file_specifier_extension(
+            module.specifier(),
+            Extension::Dts,
+          );
+          if let Some(declaration_specifier) = declaration_specifier {
+            declaration_rewrites
+              .insert(module.specifier(), declaration_specifier);
           }
+        }
+      }
+      _ => {}
+    }
+
+    to_be_rewritten.push(js);
+  }
 
-          let rewritten_path = rewrite_extension(path, Extension::Dts)
-            .unwrap_or_else(|| path.to_owned());
-          let comments = dts.comments.as_single_threaded();
-          let source_map =
-            SourceMap::single(js.specifier.clone(), (*js.source).to_owned());
-          let emitted = emit(
-            &dts.program,
-            &comments,
-            &source_map,
-            &EmitOptions {
-              source_map: deno_ast::SourceMapOption::None,
-              inline_sources: true,
-              keep_comments: true,
-            },
+  for js in to_be_rewritten {
+    let specifier_rewriter = SpecifierRewriter {
+      base_specifier: &js.specifier,
+      source_rewrites: &source_rewrites,
+      declaration_rewrites: &declaration_rewrites,
+      dependencies: &js.dependencies,
+    };
+
+    match js.media_type {
+      deno_ast::MediaType::JavaScript | deno_ast::MediaType::Mjs => {
+        let parsed_source = sources.get_parsed_source(&js.specifier).unwrap();
+        let module_info = sources
+          .analyze(&js.specifier, js.source.clone(), js.media_type)
+          .unwrap();
+        let rewritten = rewrite_specifiers(
+          parsed_source.text_info(),
+          &module_info,
+          specifier_rewriter,
+          RewriteKind::Source,
+        );
+        package_files.insert(
+          js.specifier.path().to_owned(),
+          rewritten.as_bytes().to_vec(),
+        );
+      }
+      deno_ast::MediaType::Dts | deno_ast::MediaType::Dmts => {
+        let parsed_source = sources.get_parsed_source(&js.specifier).unwrap();
+        let module_info = sources
+          .analyze(&js.specifier, js.source.clone(), js.media_type)
+          .unwrap();
+        let rewritten = rewrite_specifiers(
+          parsed_source.text_info(),
+          &module_info,
+          specifier_rewriter,
+          RewriteKind::Declaration,
+        );
+        package_files.insert(
+          js.specifier.path().to_owned(),
+          rewritten.as_bytes().to_vec(),
```

Comment:
Use `rewritten.into_bytes()` to avoid cloning and replace other occurences of `as_bytes().to_vec()`?